packages/site/src/lib/components/modals/DialectModal.svelte

```diff
@@ -0,0 +1,105 @@
+<script lang="ts">
+  import { createEventDispatcher } from 'svelte';
+  import { t } from 'svelte-i18n';
+  import Modal from 'svelte-pieces/ui/Modal.svelte';
+  import Button from 'svelte-pieces/ui/Button.svelte';
+  import { dictionary } from '$lib/stores';
+
+  export let value: string;
+  interface IAlgoliaFacetsQuery {
+    facetHits: {
+      value: string;
+    }[];
+    exhaustiveFacetsCount: boolean;
+    processingTimeMS: number;
+  }
+
+  async function fetchDialects(): Promise<IAlgoliaFacetsQuery> {
+    const response = await fetch(
+      'https://XCVBAYSYXD.algolia.net/1/indexes/entries_prod/facets/di/query',
+      {
+        method: 'POST',
+        headers: {
+          'X-Algolia-Application-Id': 'XCVBAYSYXD', // App ID
+          'X-Algolia-API-Key': 'e6d98efb32d3dc2435dce7b97ea87c3e', // Public API key
+        },
+        body: JSON.stringify({
+          facetFilters: [[`dictId:${$dictionary.id}`]],
+          maxFacetHits: 100, // Algolia max possible https://www.algolia.com/doc/api-reference/api-parameters/maxFacetHits/
+        }),
+      }
+    );
+    return await response.json();
+  }
+
+  const dispatch = createEventDispatcher<{
+    close: boolean;
+    valueupdate: {
+      field: string;
+      newValue: string;
+    };
+  }>();
+
+  const close = () => dispatch('close');
+
+  function save(newValue: string) {
+    dispatch('valueupdate', {
+      field: 'di',
+      newValue,
+    });
+    close();
+  }
+</script>
+
+<Modal on:close>
+  <span slot="heading"
+    >{$t('entry.di', {
+      default: 'Dialect',
+    })}
+  </span>
+
```

Comment:
Removed the form element since it's automatic save on hitting enter was no longer used without an input existing in this field. When copy-pasting and repurposing components, be sure to look at every line and ask if it's needed.

Reply:
Also, there was no need to disable the currently active dialect - it's fine for them to select it again and won't give confusion from the convention that a disabled button often means the option is unavailable.